it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartControllerISpec.scala

```diff
@@ -46,30 +62,90 @@ class StartControllerISpec extends BaseISpec {
+
+      sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
+    }
+
+    "store the relative continue url in the session store" in {
+      val url = "/foo"
+      implicit val request = FakeRequest(GET, s"?continue=$url")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
+    }
+
+    "store the absolute www.tax.service.gov.uk continue url in the session store" in {
+      val url = "http://www.tax.service.gov.uk/foo/bar?some=true"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
+    }
+
+    "store the whitelisted absolute external continue url in the session store" in {
+      val url = "http://www.foo.com/bar?some=false"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
 
       status(result) shouldBe 200
       bodyOf(result) should include("Create your Agent Services account")
 
       sessionStoreService.currentSession.continueUrl shouldBe Some(ContinueUrl(url))
     }
 
+    "not store the non whitelisted absolute external continue url in the session store" in {
+      val url = "http://www.foo.org/bar?some=false"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe None
+    }
+
+    "not store the absolute external continue url in the session store if the url contains an invalid character" in {
+      val url = "http://www@foo.com"
+      implicit val request = FakeRequest(GET, s"?continue=${URLEncoder.encode(url, "UTF-8")}")
+
+      val result = await(controller.start(request))
+
+      status(result) shouldBe 200
+      bodyOf(result) should include("Create your Agent Services account")
+
+      sessionStoreService.currentSession.continueUrl shouldBe None
+    }
+
     "not store the continue url in the session store if the continue param is not mentioned in the url" in {
       implicit val request = FakeRequest()
 
-      val result = await(controller.start()(request))
+      val result = await(controller.start(request))
 
       status(result) shouldBe 200
       bodyOf(result) should include("Create your Agent Services account")
 
       sessionStoreService.currentSession.continueUrl shouldBe None
     }
 
```

Comment:
Perhaps a bit too much, but we could also test for "leave an existing continue URL in the session store if a continue param is not mentioned in the url"